src/Service/TaxBenefit.php

```diff
@@ -28,6 +48,49 @@ public function calculateTaxBase() : float
         return $meterSquarePrice * $this->realEstate->getSurfaceArea();
     }
 
+    /**
+     * Calcule l'avantage fiscal en se basant sur la base fiscale et la durée
+     *
+     * @return int
+     */
+    public function calculateTaxBenefit() : float
+    {
+        $taxBenefit = 0;
+
+        $this->setTaxBase($this->calculateTaxBase());
+
+        if ($this->taxBase > self::MAXIMUM_TAX_BASE) {
+            $taxBase = self::MAXIMUM_TAX_BASE;
+        } else {
+            $taxBase = $this->taxBase;
+        }
+        if (in_array($this->getRentalPeriod(), $date = [6,9,12])) {
```

Comment:
c'est bien mieux, mais cette ligne là ne va tjs pas. déjà tu me fais une affectation dans le in_array, ce qui n'a pas de sens. Ensuite tu me remets "en dur" les valeurs 6,9,12 qu'on passe justement en constante...
je te laisse chercher un peu comment corriger ça...

Reply:
Je déclarait mon tableau dont je ne me ressert pas en dehors de cette vérification via la variable $date (qui est un tableau).
Je vais voir pour la modif.

Reply:
`if (in_array($this->getRentalPeriod(), $date = array_keys(self::TABLE_OF_RATES_BY_DURATION)) {
            $taxBenefit = $taxBase * self::TABLE_OF_RATES_BY_DURATION[$this->getRentalPeriod()];
        } else {
            throw new \LogicException("Only 6, 9, 12 accepted.");
        }

        return $taxBenefit;`
J'ai voulu test ça mais ça ne fonctionne pas...

Reply:
Corrigé.